packages/react/src/Bento/Bento.tsx

```diff
@@ -0,0 +1,298 @@
+      {React.isValidElement(TextChild) &&
+        React.cloneElement(TextChild as React.ReactElement<TextProps>, {
+          variant: TextChild.props.variant || 'muted',
+          as: TextChild.props.as || 'p',
+          size: TextChild.props.size || '300',
+          className: clsx(styles['Bento_Content-text'], TextChild.props.className),
+        })}
+      {React.isValidElement(LinkChild) &&
+        React.cloneElement(LinkChild as React.ReactElement<LinkProps>, {
+          variant: LinkChild.props.variant || 'accent',
+          className: clsx(
+            styles['Bento__call-to-action'],
+            fixedBottomLink && styles['Bento__call-to-action--fixed'],
+            LinkChild.props.className,
+          ),
+        })}
+    </div>
+  )
+}
+
+type BentoHeadingProps = BaseProps<HTMLHeadingElement> & HeadingProps
+
+const defaultHeadingTag = 'h3'
+const defaultHeadingSize = '5'
+const defaultHeadingWeight = 'medium'
+
+const _Heading = forwardRef(
+  (
+    {
+      as = defaultHeadingTag,
+      size = defaultHeadingSize,
+      weight = defaultHeadingWeight,
+      className,
+      children,
+      ...props
+    }: PropsWithChildren<BentoHeadingProps>,
+    ref: Ref<HTMLHeadingElement>,
+  ) => {
+    const childrenArray = useMemo(() => React.Children.toArray(children), [children])
+
+    const getConditionalVariant = useCallback(() => {
+      if (childrenArray.some(child => React.isValidElement(child) && child.type === 'em')) {
+        return 'muted'
+      }
+      return 'default'
+    }, [childrenArray])
+
+    const defaultColor = childrenArray.length === 1 ? 'default' : getConditionalVariant()
+
+    return (
+      <Heading
+        ref={ref}
+        className={clsx(defaultColor === 'muted' && styles[`Bento__heading--muted`], className)}
+        size={size}
+        as={as}
+        weight={weight}
+        {...props}
+      >
+        {children}
+      </Heading>
+    )
+  },
+)
+
+type BentoVisualProps = {
+  fillMedia?: boolean
+  position?: string
+  padding?: 'condensed' | 'normal' | 'spacious'
+} & React.HTMLAttributes<HTMLDivElement> &
+  BaseProps<HTMLDivElement>
+
+const Visual = ({fillMedia = true, position = '50% 50%', padding, className, children, ...rest}: BentoVisualProps) => {
+  const childrenToRender = React.Children.map(children, child => {
+    if (React.isValidElement(child)) {
+      if (child.type === 'img') {
+        return React.cloneElement(child as ReturnType<ReactHTML['img']>, {
+          style: {objectPosition: position},
+        })
+      }
+      return child
+    }
+  })
+  return (
+    <div
+      className={clsx(
+        styles.Bento__Visual,
+        !fillMedia && styles['Bento__Visual-no-fill'],
+        !!padding && styles[`Bento-padding--${padding}`],
+        className,
+      )}
+      {...rest}
+    >
+      {childrenToRender}
+    </div>
+  )
+}
+
+export const Bento = Object.assign(Root, {Item, Visual, Content, Heading: _Heading})
```

Comment:
Noticed that we have a namespaced child for headings, but. not the other content. Do we need to namespace it? I can see you're filtering the children for a heading anyway, so could we apply any default props internally to Bento.Content instead? 

![Screenshot 2023-08-31 at 09 27 36](https://github.com/primer/brand/assets/13340707/10d2338a-a941-4e77-999c-af2a3b4cd66d)

Reply:
When we spoke last and were talking about our rules for namespacing. We spoke about using namespacing when there was a significant change to the underlying element that was outside what a prop might accomplish. For instance, a className could be added by looping whereas something more complex like this would be namespaced. 

I am happy to update it if you think that will be easier to maintain, however.